apps/cowswap-frontend/src/modules/twap/containers/SetupFallbackHandlerWarning/index.tsx

```diff
@@ -0,0 +1,128 @@
+import { useSetAtom } from 'jotai/index'
+import { useEffect, useState } from 'react'
+
+import { usePrevious } from '@cowprotocol/common-hooks'
+import { ButtonPrimary, InlineBanner, Loader, BannerOrientation, UI } from '@cowprotocol/ui'
+import { useWalletInfo } from '@cowprotocol/wallet'
+
+import styled from 'styled-components/macro'
+
+import { useAllTransactions } from 'legacy/state/enhancedTransactions/hooks'
+
+import { useExtensibleFallbackContext } from '../../hooks/useExtensibleFallbackContext'
+import { useSetupFallbackHandler } from '../../hooks/useSetupFallbackHandler'
+import { verifyExtensibleFallback } from '../../services/verifyExtensibleFallback'
+import { updateFallbackHandlerVerificationAtom } from '../../state/fallbackHandlerVerificationAtom'
+
+const Banner = styled(InlineBanner)`
+  /* TODO: Make all these part of the InlineBanner props */
+  position: relative;
+  font-size: 15px;
+
+  > span {
+    gap: 20px;
+  }
+
+  > span > span {
+    gap: 20px;
+  }
+
+  &::before {
+    content: '';
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    background: var(${UI.COLOR_PAPER});
+    z-index: -1;
+    border-radius: inherit;
+  }
+
+  &::after {
+    content: '';
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    background: inherit;
+    z-index: -1;
+    border-radius: inherit;
+  }
+`
+
+const ActionButton = styled(ButtonPrimary)`
+  display: inline-block;
+  width: 100%;
+  font-size: 16px;
+  padding: 16px 24px;
+  min-height: auto;
+`
+
+export function SetupFallbackHandlerWarning() {
+  const [pendingTxHash, setPendingTxHash] = useState<string | null>(null)
+
+  const { account } = useWalletInfo()
+  const setupFallbackHandler = useSetupFallbackHandler()
+  const isTransactionPending = useIsTransactionPending(pendingTxHash)
+  const prevIsTransactionPending = usePrevious(isTransactionPending)
+  const txWasMined = prevIsTransactionPending === true && isTransactionPending === false
+
+  const updateFallbackHandlerVerification = useSetAtom(updateFallbackHandlerVerificationAtom)
+
+  const extensibleFallbackContext = useExtensibleFallbackContext()
+
+  const handleUpdateClick = async () => {
+    const txHash = await setupFallbackHandler()
+
+    if (txHash) {
+      setPendingTxHash(txHash)
```

Comment:
A re-render of this would loose the posted TX, right?

Reply:
Nice catch!
Fixed it: https://github.com/cowprotocol/cowswap/pull/5202/commits/35f1802155d78f8992ded18aa6937c44628ebac0
Thanks!